source/game_sa/Frontend/MenuManager_Draw.cpp

```diff
@@ -407,18 +413,490 @@ void CMenuManager::DrawQuitGameScreen() {
+                if (buttonText) {
+                    if (!isOppositeScreen) {
+                        CFont::PrintString(currentX, currentY, buttonText);
+                    }
+                    currentX += StretchX(75.0f);
+                    hasControl = true;
+                }
+                controlOrderIndex++;
+            }
+        }
+
+        // Handle unbound controls or special cases
+        if (!hasControl) {
+            if (controllerAction != -2) {
+                m_bRadioAvailable = 0;
+                CFont::SetColor({200, 50, 50, 255});
+                if (!isOppositeScreen) {
+                    CFont::PrintString(currentX, currentY, TheText.Get("FEC_UNB"));
+                }
+            } else {
+                CFont::SetColor({0, 0, 0, 255});
+                if (!isOppositeScreen) {
+                    CFont::PrintString(currentX, currentY, TheText.Get("FEC_CMP"));
+                }
+            }
+        }
+
+        // Handle selected action special behaviors
+        if (actionIndex == m_ListSelection) {
+            if (controllerAction == -1 || controllerAction == -2) {
+                if (actionIndex == m_ListSelection) {
+                    DisplayHelperText("FET_EIG");
+                }
+            } else {
+                m_OptionToChange = (eControllerAction)controllerAction;
+                if (m_EditingControlOptions) {
+                    if (CTimer::m_snTimeInMillisecondsPauseMode - m_lastBlinkTime > 150) {
+                        m_isTextBlinking = !m_isTextBlinking;
+                        m_lastBlinkTime = CTimer::m_snTimeInMillisecondsPauseMode;
+                    }
+                    
+                    if (m_isTextBlinking) {
+                        CFont::SetColor({0, 0, 0, 255});
+                        if (!isOppositeScreen) {
+                            CFont::PrintString(currentX, currentY, TheText.Get("FEC_QUE"));
+                        }
+                    }
+                    
+                    if (m_DeleteAllBoundControls) {
+                        DisplayHelperText("FET_CIG");
+                    } else {
+                        DisplayHelperText("FET_RIG");
+                    }
+                    m_CanBeDefined = true;
+                } else {
+                    DisplayHelperText("FET_CIG");
+                    m_CanBeDefined = false;
+                    m_DeleteAllBoundControls = false;
+                }
+            }
+        }
+
+        // Move to next line and action
+        currentY = StretchY(float(verticalOffset + (actionIndex + 1) * verticalSpacing));
+        actionIndex++;
+    }
 }
 
 // 0x57F300
 void CMenuManager::DrawControllerSetupScreen() {
-    plugin::CallMethod<0x57F300, CMenuManager*>(this);
+
+    // TODO: Magic number
+    const auto v51 = m_RedefiningControls ? 13u : (4u * (!m_ControlMethod) + 11u);
```

Comment:
also rename the variable please...
```suggestion
    const auto v51 = m_RedefiningControls 
        ? 13 
        : m_ControlMethod 
            ? 11 
            : 15;
```